Stratis.Bitcoin/BlockPulling/BlockPuller.cs

```diff
@@ -26,14 +64,41 @@ public abstract class BlockPuller : IBlockPuller
         /// <summary>Minimal quality score of a peer node based on the node's past experience with the peer node.</summary>
         public const int MinQualityScore = 1;
 
-        /// <summary>List of relations to peer nodes mapped by block header hashes that the peers are requested to provide.</summary>
-        private readonly ConcurrentDictionary<uint256, BlockPullerBehavior> map;
+        /// <summary>Instance logger.</summary>
+        protected readonly ILogger logger;
+
+        /// <summary>Lock protecting access to <see cref="map"/>, <see cref="pendingInventoryVectors"/>, <see cref="downloadedBlocks"/>, and <see cref="pendingDownloads"/></summary>
+        private readonly object lockObject = new object();
+
+        /// <summary>
+        /// List of relations to peer nodes mapped by block header hashes that the peers are requested to provide.
+        /// </summary>
+        /// <remarks>All access to this object has to be protected by <see cref="lockObject"/>.</remarks>
+        private readonly Dictionary<uint256, BlockPullerBehavior> map;
```

Comment:
I want a better name to "map" (maybe blockToPeerMappings or blocksToPeers)
Also I would suggest this text
"A list of mappings between a peer and block headers, the header represents the block that peer is asked to download"